server/src/main/java/org/candlepin/resource/OwnerResource.java

```diff
@@ -1382,17 +1825,25 @@ public JobDetail refreshPools(
         "upstream subscription, and are most commonly used for custom content delivery " +
         "in Satellite. Also helps in on-site deployment testing", value = "Create Pool")
     @ApiResponses({ @ApiResponse(code = 404, message = "Owner not found") })
-    public Pool createPool(@PathParam("owner_key") @Verify(Owner.class) String ownerKey,
-        @ApiParam(name = "pool", required = true) Pool pool) {
+    public PoolDTO createPool(@PathParam("owner_key") @Verify(Owner.class) String ownerKey,
+        @ApiParam(name = "pool", required = true) PoolDTO inputPoolDTO) {
 
-        log.info("Creating custom pool for owner {}: {}" + ownerKey, pool);
+        log.info("Creating custom pool for owner {}: {}" + ownerKey, inputPoolDTO);
 
-        // Correct owner & products
-        Owner owner = findOwner(ownerKey);
+        Pool pool = new Pool();
+
+        // Correct owner
+        Owner owner = findOwnerByKey(ownerKey);
         pool.setOwner(owner);
 
+        // Populate the rest of the pool
+        this.populateEntity(pool, inputPoolDTO);
+
+        // Resolve products
         pool = resolverUtil.resolvePool(pool);
```

Comment:
This is probably unnecessary now -- all of the resolution is done by the population of the entity from the DTO.